std/atomic.d

```diff
@@ -0,0 +1,210 @@
+/**
+ * The atomic module provides atomic struct support for lock-free
+ * concurrent programming.
+ *
+ * Copyright: Copyright Roy David Margalit 2022 - 2025.
+ * License:   $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Authors:   Roy David Margalit
+ * Source:    $(DRUNTIMESRC core/_atomic.d)
+ */
+
+module std.atomic;
+
+pragma(inline, true):
+
+/// Atomic data like std::atomic
+struct Atomic(T) if (__traits(isIntegral, T) || isPointer!T) {
+	import core.atomic : atomicLoad, atomicStore, atomicExchange, atomicFetchAdd,
+		atomicFetchSub, atomicCas = cas, atomicCasWeak = casWeak, atomicOp;
+
+	private T val;
+
+	/// Constructor
+	this(T init) shared {
+		val.atomicStore(init);
+	}
+
+	private shared(T)* ptr() shared {
+		return &val;
+	}
+
+	/// Load the value from the atomic location with SC access
+	alias load this;
+
+	/// ditto
+	T load(MemoryOrder mo = MemoryOrder.seq)() shared {
+		return val.atomicLoad!(mo.toCore);
+	}
+
+	/// Store the value to the atomic location
+	void store(MemoryOrder mo = MemoryOrder.seq)(T newVal) shared {
+		return val.atomicStore!(mo.toCore)(newVal);
+	}
+
+	/// Store using SC access
+	alias opAssign = store;
+
+	/// Atomically increment the value
+	T fadd(MemoryOrder mo = MemoryOrder.seq)(T mod) shared {
+		return atomicFetchAdd!(mo.toCore)(val, mod);
+	}
+
+	/// Atomically decrement the value
+	T fsub(MemoryOrder mo = MemoryOrder.seq)(T mod) shared {
+		return atomicFetchSub!(mo.toCore)(val, mod);
+	}
+
+	/// Atomically swap the value
+	T exchange(MemoryOrder mo = MemoryOrder.seq)(T desired) shared {
+		return atomicExchange!(mo.toCore)(&val, desired);
+	}
+
+	/// Compare and swap
+	bool cas(MemoryOrder mo = MemoryOrder.seq, MemoryOrder fmo = MemoryOrder.seq)(T oldVal, T newVal) shared {
+		return atomicCas!(mo.toCore, fmo.toCore)(ptr, oldVal, newVal);
+	}
+
+	/// ditto
+	bool casWeak(MemoryOrder mo = MemoryOrder.seq, MemoryOrder fmo = MemoryOrder.seq)(T oldVal,
+			T newVal) shared {
+		return atomicCasWeak!(mo.toCore, fmo.toCore)(ptr, oldVal, newVal);
+	}
+
+	/// Op assign with SC semantics
+	T opOpAssign(string op)(T rhs) shared {
+		return val.atomicOp!(op ~ `=`)(rhs);
+	}
+
+	/// Implicit conversion to FADD and FSUB
+	T opUnary(string op)() shared if (op == `++`) {
+		return val.atomicOp!`+=`(1);
+	}
+
+	T opUnary(string op)() shared if (op == `--`) {
+		return val.atomicOp!`-=`(1);
+	}
+
+	auto ref opUnary(string op)() shared if (op == `*`) {
+		return *(load);
+	}
+}
+
+private alias TestAtomic = Atomic!(int);
```

Comment:
If this is only for testing,  `version(unittest)`?

Reply:
Removed